lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1914,7 +1930,7 @@ export class UserAgentApplication {
               parameters["error_description"] = "Invalid idToken. idToken: " + parameters[Constants.idToken];
               this.cacheStorage.setItem(Constants.msalError, "invalid idToken");
               this.cacheStorage.setItem(Constants.msalErrorDescription, "Invalid idToken. idToken: " + parameters[Constants.idToken]);
-              error = ClientAuthError.createInvalidIdTokenError(response.idToken);
+              error = ClientAuthError.createInvalidIdTokenError(response.getIdToken());
```

Comment:
I think if you got rid of all the getters and setters you could just figure out your type and varaibles and then create an object of type

Reply:
The getters and setters behave functionally the same as a type and variables would. The main benefit of using a class is that instantiation is easier. Using a type would require knowing the values beforehand and setting them at time of creation. With a class we can encapsulate this variable setting and create defaults outside of the UAA class, while keeping the same functionality of a type.

Reply:
I'm happy to revisit this if needed, but that was my justification for using class over type.